HSLdevcom/jore-map-ui
# Create new node tool

Closes #604

BE PR: https://gitlab.hsl.fi/jore/jore-map-backend/merge_requests/31

* moved isSelected to nodeMarker props to solve problem where at node/new node was not being selected (now it is because we can say node is selected by checking if url has subSites.newNode)

* moved isSelected to nodeMarker props to solve problem where at node/new node was not being selected (now it is because we can say node is selected by checking if url has subSites.newNode)

* Centering map is now EditNodeLayer's responsibility --> map is centered after new node is created

* Fixed bug of node change being called multiple times

* "Network should not refresh when dragging node"
--> this is how our vectorGridLayer works at the moment. We have "// TODO: find a better way to achieve this". Didn't make a new card to fix this because lag issue is not fixed, it was because our node was changed 9 times due to bugs in NodeCoordinatesView

TODO later:
After node is created, redirect to node/id: CARD: #666 

How to test:

1. Create a new node
1. After save, backend console should log new node id
1. Paste new node id to url: http://localhost:3000/node/{nodeId}
